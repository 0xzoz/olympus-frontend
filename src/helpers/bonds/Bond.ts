import { Token } from "src/helpers/contracts/Token";

import { LPToken } from "../contracts/LPToken";

export interface BondConfig {
  id: string;
  baseToken: Token;
  quoteToken: Token | LPToken;
}

export class Bond {
  /**
   * Bond market id
   */
  id: BondConfig["id"];

  /**
   * The token the market buys from the protocol
   */
  baseToken: BondConfig["baseToken"];

  /**
   * The token that the market sells to the protocol
   */
  quoteToken: BondConfig["quoteToken"];

  constructor(config: BondConfig) {
    this.id = config.id;
    this.baseToken = config.baseToken;
    this.quoteToken = config.quoteToken;
  }
}
