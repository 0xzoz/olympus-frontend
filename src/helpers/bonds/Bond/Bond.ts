import { Token } from "src/helpers/contracts/Token/Token";

export interface BondConfig {
  baseToken: Token;
  quoteToken: Token;
}

export class Bond {
  /**
   * The token the user buys from the protocol
   */
  baseToken: Token;

  /**
   * The token that the user sells to the protocol
   */
  quoteToken: Token;

  constructor(config: BondConfig) {
    this.quoteToken = config.quoteToken;
    this.baseToken = config.baseToken;
  }
}
