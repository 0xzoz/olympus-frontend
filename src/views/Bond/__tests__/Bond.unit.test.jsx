import { render } from "src/testUtils";

import { Bond } from "../Bond";

describe("<ChooseBond/>", () => {
  it("should render component", () => {
    const { container } = render(<Bond />);
    expect(container).toMatchSnapshot();
  });
});
