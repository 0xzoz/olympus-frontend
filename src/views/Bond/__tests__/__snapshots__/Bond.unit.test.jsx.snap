// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseBond/> should render component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiPaper-root makeStyles-root-2 makeStyles-root-3  MuiPaper-elevation0 MuiPaper-rounded"
      style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root card-header MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-4"
          >
            <div
              class="MuiBox-root MuiBox-root-5"
            >
              <h5
                class="MuiTypography-root header-text MuiTypography-h5"
              >
                Bond (4,4)
              </h5>
            </div>
            <div
              class="top-right"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-end"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="makeStyles-root-7 "
                >
                  <div
                    class="MuiBox-root MuiBox-root-8"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
                    >
                      Treasury Balance
                    </h5>
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                      style="width: 100%;"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                        style="width: 100%;"
                      />
                    </h4>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="makeStyles-root-7 "
                >
                  <div
                    class="MuiBox-root MuiBox-root-9"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
                    >
                      OHM Price
                    </h5>
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                      style="width: 100%;"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                        style="width: 100%;"
                      />
                    </h4>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-10"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
